Firmware/ultralcd.cpp

```diff
@@ -7537,7 +7526,6 @@ void lcd_ignore_click(bool b)
 
 void lcd_finishstatus() {
   SERIAL_PROTOCOLLNRPGM(MSG_LCD_STATUS_CHANGED);
-  lcd_padstatus();
```

Comment:
Doesn’t this make the `lcd_finishstatus()` function a bit useless? It used to pad the status line if the message was not 20 characters. 

I think its better to do it here than in `lcd_ui_print_status_line()`, because then we aren’t padding the message multiple times with each print on the LCD. Before we only needed to pad the mesage once when `lcd_status_message` was set.

Reply:
Indeed. I moved the only relevant part inside `lcd_updatestatus()` itself.

The only difference to before is that in `check_file()` we were forcing a full display clear, but by testing this it doesn't look like it's needed at all (@leptun was this really necessary?). I get a clean screen after the file check, even if I have the additional warning messages due to incompatible gcode.

The issue is that we need to compare if the message we're going to display is newer or not and compare it to the current. The new message is not padded, so we would need a custom comparison function to ignore (but check) the extra padding or extra memory to pre-pad the new string and compare it. I didn't test this, but I suspect neither would save space.

I don't think lcd_print_pad is much slower than before (it's not a critical path anyway). Using lcd_print_pad() also fixes two other cases (PID autotune) that were displaying a custom status message without padding so could show something in the background.

Reply:
I don't remember why that was needed. If it works now, then it can be removed. It might have been a relic of the past that I just kept anyway

Reply:
@3d-gussner when you test this, also check if after the "checking file" step the screen is fine. It looks like it always work correctly to me.